app/controllers/profiles_controller.rb

```diff
@@ -19,6 +22,12 @@ def show
 
   private
 
+  def authorize_own_profile
+    if current_user.id != session[:user_id]
+      redirect_to login_url, alert: "Not authorized" 
+    end
+  end
+
   def current_user
     @user ||= User.find(params[:user_id])
```

Comment:
If you use `find_by` it will return nil when no use is found and the check above will work